File: beginner/109/C.py
def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return(a)

N, K = map( int, input().split())
X = list( map( int, input().split()))
now = abs( K - X[0])
for x in X:
    now = gcd(now, abs(x-K))
print(now)

File: beginner/109/C.cpp
#include <iostream>
#include <vector>
#include <cstdio>
#include <string>
#include <algorithm>
#include <cstdlib>
using namespace std;
typedef long long int ll;
#define all(x) x.begin(),x.end()
int main()
{
  int N, t, ans, s;
  cin >> N >> t;
  int x[N];
  for (int i = 0; i < N; ++i) {
    cin >> x[i];
  }
  ans = abs(t-x[0]);
  for (int i=0; i < N; ++i) {
    s = ans;
    ans = __gcd(s, abs(t-x[i]));
  }
  cout << ans << "\n";
  return 0;
}
